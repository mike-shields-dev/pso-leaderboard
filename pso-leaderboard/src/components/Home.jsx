import React from "react";
import { Link } from "react-router-dom";
import style from "../styles/Home.module.css";
import PageTitleSetter from "./PageTitleSetter";

const Home = () => {
  return (
    <>
      <PageTitleSetter title="Home" />

      <div className={style.home}>
        <p>Home</p>
        <nav>
          <Link to="/display">Display</Link>
          <Link to="/editor">Editor</Link>
        </nav>
      </div>
    </>
  );
};

export default Home;
