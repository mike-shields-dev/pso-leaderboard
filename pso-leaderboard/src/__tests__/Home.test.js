import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Home from "../components/Home";

const renderHome = () => render(<Home />, { wrapper: BrowserRouter });

describe("Home", () => {
  it("matches snapshot", () => {
    const { asFragment } = renderHome();

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders hyperlinks to '/display' and '/editor'", () => {
    renderHome();

    expect(screen.getByRole("link", { name: /display/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /editor/i })).toBeInTheDocument();
  });
});
