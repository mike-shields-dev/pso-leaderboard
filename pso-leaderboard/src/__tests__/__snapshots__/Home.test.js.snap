// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <p>
      Home
    </p>
    <nav>
      <a
        href="/display"
      >
        Display
      </a>
      <a
        href="/editor"
      >
        Editor
      </a>
    </nav>
  </div>
</DocumentFragment>
`;
